aioshelly/wsrpc.py

```diff
@@ -99,6 +125,44 @@ async def connect(self, aiohttp_session: aiohttp.ClientSession) -> None:
 
         _LOGGER.info("Connected to %s", self._ip_address)
 
+        # https://shelly-api-docs.shelly.cloud/gen2/Overview/CommonDeviceTraits/#authentication-over-websocket
+        if options and options.password:
+            try:
+                # First API call should fail with 401 if auth enabled
+                await self.call("Sys.GetStatus", {})
```

Comment:
I think the `WsRPC` class should abstain from invoking specific methods on the device -- it should focus on purely providing the RPC channel. The authentication flow can happen for the first (or every?) call which gets a 401. That way, `WsRPC` doesn't need to know about methods and their semantics, and will do the authentication step only when necessary.

Reply:
That is exactly what I was looking for, similar what we do for Gen1, if we have auth we should just pass it on every call not trying to fail first and retry, this with the example how to make the first auth https://github.com/home-assistant-libs/aioshelly/pull/192#pullrequestreview-899520621 would provide a clean solution.
@figadore let me know if you plan to work on this changes, otherwise I can try to pick it up and we commit it together.